ovirtlib4/system_service.py

```diff
@@ -64,8 +70,94 @@ def _get_collection_entity(self):
         """
         return CollectionEntity(connection=self.connection)
 
-    def get(self, *args, **kwargs):
-        """ Same as list() """
+    def _process_sample(self, wait_for, wait_method, sample):
+        """
+        Handle a result sample according to the given 'wait_for' and 'wait_method'
+
+        Args:
+            wait_for (object): see run_sampler() docstring
+            wait_method (str): see run_sampler() docstring
+            sample (list): List of SDK entities, returned by get()
+
+        Returns:
+            list: If 'str' or 'bool' return the given sample if it match,
+                if 'str' return the filtered list if it match,
+                if no match return an empty list
+        """
+        if type(wait_for) == bool:
+            if (sample is not []) == wait_for:
```

Comment:
you define sample as a list in the docstring:
            sample (list): List of SDK entities, returned by get()

Reply:
Here we check if the sample (that it's a list) is empty or not 
and compared it to 'wait_for' that expected to be True or False.

Reply:
so maybe:
if (not sample) == wait_for:

Reply:
To be equivalent you need to write 
if (not sample) == (not wait_for):

But I prefer to write: "sample is not []"
It easy to understand, it's like saying: "if sample is not an empty list"

Reply:
np